### PR TITLE
feat: [ANRDROAPP-2733] Change form alerts design to bottomSheetDialogs

### PR DESCRIPTION
**Pull request by mmmateos**
Jira Issue: https://jira.dhis2.org/browse/ANDROAPP-2733

### CODE REVIEW

File: app/src/main/java/org/dhis2/usescases/enrollment/EnrollmentActivity.kt [hunk 1]
```diff
@@ -124,7 +123,9 @@ class EnrollmentActivity : ActivityGlobalAbstract(), EnrollmentView {
         binding.fieldRecycler.adapter = adapter
 
         binding.next.setOnClickListener {
-            if (presenter.dataIntegrityCheck(adapter.mandatoryOk(), adapter.hasError())) {
+            if (
+                presenter.dataIntegrityCheck(adapter.emptyMandatoryFields(), adapter.errorFields())
+            ) {
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/enrollment/EnrollmentActivity.kt]: can check the code format ?

File: app/src/main/java/org/dhis2/utils/customviews/AlertBottomDialog.kt [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        this.showNegative = true
+        this.negativeText = text
+        this.negativeOnclick = onClick
+        return this
+    }
+
+    fun setEmptyMandatoryFields(emptyMandatoryFields: List<String>): AlertBottomDialog {
+        this.emptyMandatoryFields = emptyMandatoryFields
+        return this
+    }
```
Comment (ferdyrod) [inline app/src/main/java/org/dhis2/utils/customviews/AlertBottomDialog.kt]: All this setXXXX methods can benefit from the _.apply{ }_ high-order function.
```kotlin
fun setMessage(message: String) = apply { this.message = message }

fun setEmptyMandatoryFields(fields: List<String>) = apply { this.emptyMandatoryFields = emptyMandatoryFields }
```

**Comment by ferdyrod** [comment]
Tested on a device!

File: app/src/main/java/org/dhis2/utils/customviews/AlertBottomDialog.kt (hunk 2 above)
Comment (mmmateos) [reply app/src/main/java/org/dhis2/utils/customviews/AlertBottomDialog.kt]: Nice, ok :+1:

**Pull request merged**